leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -312,20 +321,100 @@ public static void main(final String[] args) {
                 return;
             }
         }
+
         // Set boolean controlling OSCORE usage
-        boolean useOSCore = false;
+        OSCoreSettings oscoreSettings = null;
         if (cl.hasOption("oscore")) {
             System.out.println("Using OSCORE");
-            useOSCore = true;
 
             HashMapCtxDB db = OscoreHandler.getContextDB();
             OSCoreCoapStackFactory.useAsDefault(db);
 
+            // Parse OSCORE related command line parameters
+
+            String mastersecretStr = cl.getOptionValue("mastersecret");
+            String mastersaltStr = cl.getOptionValue("mastersalt");
+            String idcontextStr = cl.getOptionValue("idcontext");
+            String senderidStr = cl.getOptionValue("senderid");
+            String recipientidStr = cl.getOptionValue("recipientid");
+            String aeadStr = cl.getOptionValue("aead");
+            String hkdfStr = cl.getOptionValue("hkdf");
+
+            if (mastersecretStr == null) {
+                System.err.println("The OSCORE master secret must be indicated");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Set salt to empty string if not indicated
+            if (mastersaltStr == null) {
+                mastersaltStr = "";
+            }
+
+            if (idcontextStr != null) {
+                System.err.println("The OSCORE ID Context parameter is not yet supported");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            if (senderidStr == null) {
+                System.err.println("The OSCORE Sender ID must be indicated");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Check that the Sender ID matches the client name
+            // http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Transport-V1_1-20180710-A.html#5-5-5-0-555-Endpoint-Client-Name
+            if (!Arrays.equals(endpoint.getBytes(), Hex.decodeHex(senderidStr.toCharArray()))) {
```

Comment:
If senderID must match the endpoint name, we can remove the senderID option and just document in help that endpoint client name will be used as senderID ?

Reply:
After discussions with Ericsson it may be better to remove this check. It is better if the Sender ID can be kept short since it will be included in every outgoing request. So having a long client endpoint name can then induce a lot of communication overhead. Thus it can be better to not require that they match.

I removed this check in commit 3892cb62eaadfb62acf3e7bcf1469952b05dae4b